checker/src/org/checkerframework/checker/index/searchindex/SearchIndexAnnotatedTypeFactory.java

```
@@ -0,0 +1,106 @@
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(
+            MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+        return new SearchIndexQualifierHierarchy(factory);
+    }
+
+    private final class SearchIndexQualifierHierarchy extends MultiGraphQualifierHierarchy {
+
+        public SearchIndexQualifierHierarchy(
+                MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+            super(factory);
+        }
+
+        @Override
+        public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
+            if (AnnotationUtils.areSameByClass(superAnno, SearchIndexUnknown.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, SearchIndexBottom.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, SearchIndexUnknown.class)) {
+                return false;
+            }
+            if (AnnotationUtils.areSameByClass(superAnno, SearchIndexBottom.class)) {
+                return false;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, SearchIndex.class)) {
+                if (AnnotationUtils.areSameByClass(superAnno, NegativeIndexFor.class)) {
+                    return false;
+                }
+                return IndexUtil.getValueOfAnnotationWithStringArgument(superAnno)
+                        .containsAll(IndexUtil.getValueOfAnnotationWithStringArgument(subAnno));
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, NegativeIndexFor.class)) {
+                if (AnnotationUtils.areSameByClass(superAnno, SearchIndex.class)) {
+                    return true;
+                }
+                return IndexUtil.getValueOfAnnotationWithStringArgument(superAnno)
+                        .containsAll(IndexUtil.getValueOfAnnotationWithStringArgument(subAnno));
+            }
+            return false; // dead code
```

If it's dead code, throw an exception rather than silently returning a value.